test-app/runtime/src/main/cpp/ObjectManager.cpp

```diff
@@ -341,6 +341,10 @@ void ObjectManager::JSObjectFinalizer(Isolate *isolate, ObjectWeakCallbackState
         auto jsInfoIdx = static_cast<int>(MetadataNodeKeys::JsInfo);
         po->Get(m_isolate)->SetInternalField(jsInfoIdx, Undefined(m_isolate));
         po->Reset();
+        auto it = m_idToObject.find(javaObjectID);
+        m_idToObject.erase(it);
```

Comment:
It would be easier and safer to directly `m_idToObject.erase(javaObjectID);` here.